### ISSUE TITLE
Joken token constructed from struct unable to pick JSON decoder

### ISSUE THREAD

**Issue by JonRowe** [bryanjos/joken#183]
In the documentation the basic recommended verifier function for the plug doesn't work when you have a JSON payload (criteria?).

```Elixir
def verify_function() do
  %Joken.Token{}
  |> Joken.with_signer(hs256("secret"))
end

compact_token = "...string generated with json payload..."

# This is what the plug essentially does to validate a token
verify_function()
|> with_compact_token(compact_token)
|> verify

# %Joken.Token{
#   error: "Could not verify token",
#   ...
# }
```

By contrast constructing a token from a compacted token will match and verify e.g.

```Elixir
import Joken

compact_token = "...string generated with json payload..."
compacted_token
|> joken
|> with_signer

# %Joken.Token{
#   claims: %{"json" => "data"},
#   error: nil
#   ...
# }
```

The difference is in the first example you have to specify a JSON decoder for it to match (shout out to #176 for that info), it is automatically detected however in the second.

I believe this is confusing behaviour, if we can correctly figure out we have a json payload from a compacted token we should be able to figure it out when using a token as a verifier. The logic of loading a compacted token in should be the same!

At the very least we should update the documentation to show you must set the JSON decoder manually when using the Plug.

**Comment by victorolinasc** [comment]
This is because we did not provide a default Poison value for the decoder. We will be changing to Jason as a default decoder on version 2 and will make sure it is set no matter how you configure Joken.

I'll update the README to make this explicit though while we don't release a version 2.

Thanks for your feedback. I'll close this one when the README is updated.

**Comment by victorolinasc** [comment]
Fixed with d934340d58fe3e69f3d8da88eef532f588b15786
